src/API/CommandBus.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/* Copyright (c) 1998-2019 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\API;
+
+use ILIAS\API\Int\AbstractCommandHandler;
+use ILIAS\API\Exceptions as Exc;
+use ILIAS\API\Int\CommandHandler;
+
+/**
+ *
+ *
+ * @author killing@leifos.de
+ */
+class CommandBus
+{
+	/**
+	 * Constructor
+	 */
+	public function __construct()
+	{
+		// should be injected
+		$this->handler_factory = new \ILIAS\API\HandlerFactory();
+	}
+
+	/**
+	 * Dispatch command, checks policy on all handlers
+	 * that correspond to the involved factories.
+	 *
+	 * Parent handlers can configure subsequent command handlers.
+	 *
+	 * The final command handler checks the policy on the command and handles it.
+	 *
+	 * @param Int\Command $command
+	 * @param int $actor_id
+	 * @throws Exc\PolicyFailed
+	 */
+	public function dispatch(Int\Command $command, int $actor_id)
```

Comment:
When first reading this I had some pain to understand this, but know I got this!